### DISCUSSION TITLE
zh/guides/distance-calculation/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#15746] [Docs]
# zh/guides/distance-calculation/

了解如何使用Ultralytics YOLOv8  计算物体之间的距离，以实现准确的空间定位和场景理解。

https://docs.ultralytics.com/zh/guides/distance-calculation/

**Comment by zgr1446741194** [comment]
Hello author, I would like to propose a scenario similar to this: using monocular distance measurement to calculate the distance from the distant water level line (similar to the shoreline) in the video stream obtained by the camera to my camera (a kind of early warning of the water level), how can I use the solution? At present, the difficulty can be determined that we need to complete distance conversion similar to 3D to 2D, because of perspective space, instead of using fixed linear distance conversion similar to visionEye (from pixels to realistic distance). Another difficulty is that we do not know the true physical properties of the observed water line, which means that we cannot use the traditional method of monocular distance measurement to deduce the distance from the true length or height of the observed object. If you have any viable solutions, thank you very much for adding the code demo in your reply.

**Reply by glenn-jocher** [reply]
For calculating distances to a waterline using monocular vision, consider using depth estimation techniques or stereo vision for more accurate results. YOLO models can assist in object detection, but for precise distance measurement, integrating additional sensors or methods might be necessary. For more details, please refer to our [distance calculation guide](https://docs.ultralytics.com/guides/distance-calculation/).
